runtime-modules/working-group/src/lib.rs

```diff
@@ -369,10 +372,36 @@ decl_module! {
 
             checks::ensure_valid_reward_per_block::<T, I>(&reward_per_block)?;
 
+            // Lead needs stake to generate
+            if opening_type == OpeningType::Regular {
+                // Lead must be set for ensure_origin_for_openig_type in the
+                // case of regular.
+                let lead = Self::worker_by_id(Self::current_lead().unwrap());
+
+                ensure!(
+                    T::StakingHandler::is_enough_balance_for_stake(
+                        &lead.staking_account_id,
+                        T::OpeningStake::get()
+                    ),
+                    Error::<T, I>::InsufficientBalanceToCoverStake
+                );
+            }
+
             //
             // == MUTATION SAFE ==
             //
 
+            if opening_type == OpeningType::Regular {
+                // Lead must be set for ensure_origin_for_openig_type in the
+                // case of regular.
+                let lead = Self::worker_by_id(Self::current_lead().unwrap());
```

Comment:
There should not be any `unwrap()` or `expect()` calls in the chain-code (only in tests). `ensure_origin_for_openig_type()` validates leader existense internally, however in case of code changes `unwrap()` could backfire. Consider creating some `ensure_leader_exist` function that returns leader along with a `DispatchError` similar to `ensure_worker_exists`.